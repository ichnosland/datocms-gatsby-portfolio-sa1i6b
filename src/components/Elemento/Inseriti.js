/**
*
* Inseriti
*
*/

import styled, { css } from 'styled-components';
import { transparentize } from 'polished';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const Inseriti = styled.div`
  min-height: 34px;
  padding: 10px;
  margin: 20px auto;
  background-color: ${colore.ui.contrast};
  border-radius: 3px;
  ${media.desktop`
    min-height: 41px;
    border: 1px solid ${transparentize(0.5, colore.ui.darkBg)};
    ${({ giusto }) => giusto && css`
      border: 2px solid ${colore.actions.okay};
    `};
    ${({ sbagliato }) => sbagliato && css`
      border: 2px solid ${colore.actions.error};
    `};
  `};
  ${({ giusto }) => giusto && css`
    border: 2px solid ${colore.actions.okay};
  `};
  ${({ sbagliato }) => sbagliato && css`
    border: 2px solid ${colore.actions.error};
  `};
`;
