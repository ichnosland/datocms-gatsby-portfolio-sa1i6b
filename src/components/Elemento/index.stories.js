import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
import { ThemeProvider } from 'styled-components';
import { allThemes, themeOptions } from 'style/theme';
import Elemento from './index';
import { Inseriti } from './Inseriti';

const dragColors = [
  'uno',
  'due',
  'tre',
  'quattro',
  'cinque',
];

const nota = `
  La props 'disabled' si limita a togliere il pointer-events.\n
  Il componente sembra un blocco ma in realtà è un FlexBox. È probabile che in futuro diventi inline-flexbox.
`;

storiesOf('Elemento', module)
  .addDecorator(withKnobs)
  .add('Elemento', withNotes(nota)(() => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <Elemento
        dragColor={select('dragColor', dragColors, 'uno')}
        drag={boolean('drag', false)}
        dragDisabled={boolean('dragDisabled', false)}
        label={text('label', 'Testo elemento')}
        dragLabel={text('dragLabel', 'Testo etichetta')}
        inattivo={boolean('inattivo', false)}
        trovato={boolean('trovato', false)}
        giusto={boolean('giusto', false)}
        sbagliato={boolean('sbagliato', false)}
        inserito={boolean('inserito', false)}
        disabled={boolean('disabled', false)}
      />
    </ThemeProvider>
  )))
  .add('Elemento Inserito', () => (
    <Inseriti
      giusto={boolean('giusto', false)}
      sbagliato={boolean('sbagliato', false)}
    >
      <Elemento
        inserito
        label="elemento inserito"
      />
    </Inseriti>
  ));
