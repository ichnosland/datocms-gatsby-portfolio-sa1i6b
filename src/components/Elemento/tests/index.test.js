import React from 'react';
import { shallow } from 'enzyme';

import FlexBox from 'components/FlexBox';
import Elemento, {
  BaseElemento,
  SubLabel,
} from '../index';

const label = 'myLabel';
const dragLabel = 'myDragLabel';

const theme = {
  brand: '#111111',
  light: '#333333',
  subtle: '#ffffff',
};

const dragColors = [
  '#f5ee97',
  '#d6f4a0',
  '#E0D5E3',
  '#ffd8bf',
  '#cde5ff',
];

describe('<Elemento />', () => {
  it('should render BaseElemento', () => {
    const renderedComponent = shallow(<Elemento label="ciao" />);
    expect(renderedComponent.find(BaseElemento).length).toEqual(1);
  });

  it('should render the props label', () => {
    const renderedComponent = shallow(<Elemento label={label} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the props dragLabel', () => {
    const renderedComponent = shallow(<Elemento label={label} dragLabel={dragLabel} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props inattivo', () => {
    const renderedComponent = shallow(<Elemento label={label} inattivo />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props disabled', () => {
    const renderedComponent = shallow(<Elemento label={label} disabled />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props drag', () => {
    const renderedComponent = shallow(<Elemento label={label} drag />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props dragColor', () => {
    const renderedComponent = shallow(<Elemento label={label} drag dragColor={dragColors[3]} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props dragDisabled', () => {
    const renderedComponent = shallow(<Elemento label={label} drag dragDisabled />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props trovato', () => {
    const renderedComponent = shallow(<Elemento label={label} drag trovato />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props giusto', () => {
    const renderedComponent = shallow(<Elemento label={label} drag giusto />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props sbagliato', () => {
    const renderedComponent = shallow(<Elemento label={label} drag sbagliato />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its props inserito', () => {
    const renderedComponent = shallow(<Elemento label={label} inserito />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props onClickFx', () => {
    const renderedComponent = shallow(<Elemento label={label} onClickFx={() => {}} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<BaseElemento />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<BaseElemento />);
    expect(renderedComponent.find(FlexBox).length).toEqual(1);
  });

  it('should render the exactLabel', () => {
    const renderedComponent = shallow(<BaseElemento label={label} />);
    expect(renderedComponent.prop('label')).toEqual('myLabel');
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props theme', () => {
    const renderedComponent = shallow(<BaseElemento theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props inattivo', () => {
    const renderedComponent = shallow(<BaseElemento inattivo />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props disabled', () => {
    const renderedComponent = shallow(<BaseElemento disabled />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props trovato', () => {
    const renderedComponent = shallow(<BaseElemento trovato />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props drag', () => {
    const renderedComponent = shallow(<BaseElemento drag />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dragDisabled', () => {
    const renderedComponent = shallow(<BaseElemento drag dragDisabled />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props giusto', () => {
    const renderedComponent = shallow(<BaseElemento giusto />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props sbagliato', () => {
    const renderedComponent = shallow(<BaseElemento sbagliato />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props drag + giusto', () => {
    const renderedComponent = shallow(<BaseElemento drag giusto />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props drag + sbagliato', () => {
    const renderedComponent = shallow(<BaseElemento drag sbagliato />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props inserito', () => {
    const renderedComponent = shallow(<BaseElemento inserito />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<SubLabel />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<SubLabel />);
    expect(renderedComponent.find('Div').length).toEqual(1);
  });

  it('should render its css w/ props disabled', () => {
    const renderedComponent = shallow(<SubLabel disabled />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dragDisabled', () => {
    const renderedComponent = shallow(<SubLabel dragDisabled />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dragColor uno', () => {
    const renderedComponent = shallow(<SubLabel dragColor="uno" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dragColor due', () => {
    const renderedComponent = shallow(<SubLabel dragColor="due" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dragColor tre', () => {
    const renderedComponent = shallow(<SubLabel dragColor="tre" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dragColor quattro', () => {
    const renderedComponent = shallow(<SubLabel dragColor="quattro" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dragColor cinque', () => {
    const renderedComponent = shallow(<SubLabel dragColor="cinque" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
