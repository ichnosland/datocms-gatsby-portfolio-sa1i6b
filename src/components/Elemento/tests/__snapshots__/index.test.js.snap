// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseElemento /> should render its css w/ props disabled 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  disabled={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props drag + giusto 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: rgb(255,255,255);
  border-width: 2px;
  border-style: solid;
  box-shadow: none;
  pointer-events: none;
  background-color: rgb(153,226,17);
  border-color: rgb(153,226,17);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgb(78,78,78);
  background-color: rgb(255,255,255);
  border-width: 2px;
  border-style: solid;
  box-shadow: none;
  pointer-events: none;
  border-color: rgb(153,226,17);
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  drag={true}
  giusto={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props drag + sbagliato 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: rgb(255,255,255);
  border-width: 2px;
  border-style: solid;
  box-shadow: none;
  pointer-events: none;
  background-color: rgb(255,0,0);
  border-color: rgb(255,0,0);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgb(78,78,78);
  background-color: rgb(255,255,255);
  border-width: 2px;
  border-style: solid;
  box-shadow: none;
  pointer-events: none;
  border-color: rgb(255,0,0);
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  drag={true}
  sbagliato={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props drag 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  drag={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props dragDisabled 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  pointer-events: none;
  box-shadow: none;
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0 span {
  color: rgb(209,209,209);
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  drag={true}
  dragDisabled={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props giusto 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: rgb(255,255,255);
  border-width: 2px;
  border-style: solid;
  box-shadow: none;
  pointer-events: none;
  background-color: rgb(153,226,17);
  border-color: rgb(153,226,17);
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  giusto={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props inattivo 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  opacity: 0.3;
  box-shadow: none;
  cursor: default;
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:hover,
.c0:active,
.c0:focus {
  opacity: 0.3;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  inattivo={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props inserito 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: none;
  border: none;
  margin-bottom: 0;
  box-shadow: none;
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0 span {
  padding: 0;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: rgb(78,78,78);
  background: none;
  border: none;
  box-shadow: none;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #00BBEF;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  inserito={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props sbagliato 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: rgb(255,255,255);
  border-width: 2px;
  border-style: solid;
  box-shadow: none;
  pointer-events: none;
  background-color: rgb(255,0,0);
  border-color: rgb(255,0,0);
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  sbagliato={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props theme 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #111111;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #111111;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  theme={
    Object {
      "brand": "#111111",
      "light": "#333333",
      "subtle": "#ffffff",
    }
  }
/>
`;

exports[`<BaseElemento /> should render its css w/ props trovato 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: rgb(255,255,255);
  background-color: #C9E4FF;
  border-color: #C9E4FF;
  box-shadow: none;
  cursor: default;
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:hover,
.c0:active,
.c0:focus {
  border-color: #C9E4FF;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
  trovato={true}
/>
`;

exports[`<BaseElemento /> should render the exactLabel 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgb(78,78,78);
  margin: 0 6px 8px 0;
  background-color: rgb(255,255,255);
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 2px;
  box-shadow: 0 0 1px rgba(0,0,0,0.1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover {
  border-color: rgba(0,0,0,0.4);
}

.c0:active {
  color: #fff;
  background-color: #00BBEF;
  box-shadow: none;
}

.c0:active div {
  color: #fff;
  background-color: #00BBEF;
}

.c0 span {
  padding: 6px 8px;
}

.c0 span > span {
  padding: 0;
}

.c0:last-child {
  margin-right: 0;
}

<FlexBox
  className="c0"
  label="myLabel"
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
/>
`;

exports[`<Elemento /> should render its css w/ props onClickFx 1`] = `
<Elemento__BaseElemento
  onClick={[Function]}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props disabled 1`] = `
<Elemento__BaseElemento
  disabled={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props drag 1`] = `
<Elemento__BaseElemento
  drag={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <Elemento__SubLabel>
    <span />
  </Elemento__SubLabel>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props dragColor 1`] = `
<Elemento__BaseElemento
  drag={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <Elemento__SubLabel
    dragColor="#ffd8bf"
  >
    <span />
  </Elemento__SubLabel>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props dragDisabled 1`] = `
<Elemento__BaseElemento
  drag={true}
  dragDisabled={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <Elemento__SubLabel
    dragDisabled={true}
  >
    <span />
  </Elemento__SubLabel>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props giusto 1`] = `
<Elemento__BaseElemento
  drag={true}
  giusto={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <Elemento__SubLabel>
    <span />
  </Elemento__SubLabel>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props inattivo 1`] = `
<Elemento__BaseElemento
  inattivo={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props inserito 1`] = `
<Elemento__BaseElemento
  inserito={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props sbagliato 1`] = `
<Elemento__BaseElemento
  drag={true}
  sbagliato={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <Elemento__SubLabel>
    <span />
  </Elemento__SubLabel>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render its props trovato 1`] = `
<Elemento__BaseElemento
  drag={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
  trovato={true}
>
  <Elemento__SubLabel>
    <span />
  </Elemento__SubLabel>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render the props dragLabel 1`] = `
<Elemento__BaseElemento
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<Elemento /> should render the props label 1`] = `
<Elemento__BaseElemento
  theme={
    Object {
      "brand": "#00BBEF",
      "light": "#C9E4FF",
    }
  }
>
  <span>
    myLabel
  </span>
</Elemento__BaseElemento>
`;

exports[`<SubLabel /> should render its css w/ props disabled 1`] = `
.c0 {
  display: block;
  width: 100%;
  min-height: 18px;
  color: rgb(96,96,96);
  padding-top: 3px;
  background-color: rgb(240,240,240);
  font-family: sans-serif;
  font-size: 11px;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<Div
  className="c0"
  disabled={true}
/>
`;

exports[`<SubLabel /> should render its css w/ props dragColor cinque 1`] = `
.c0 {
  display: block;
  width: 100%;
  min-height: 18px;
  color: rgb(96,96,96);
  padding-top: 3px;
  background-color: rgb(240,240,240);
  font-family: sans-serif;
  font-size: 11px;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #cde5ff;
}

<Div
  className="c0"
  dragColor="cinque"
/>
`;

exports[`<SubLabel /> should render its css w/ props dragColor due 1`] = `
.c0 {
  display: block;
  width: 100%;
  min-height: 18px;
  color: rgb(96,96,96);
  padding-top: 3px;
  background-color: rgb(240,240,240);
  font-family: sans-serif;
  font-size: 11px;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #d6f4a0;
}

<Div
  className="c0"
  dragColor="due"
/>
`;

exports[`<SubLabel /> should render its css w/ props dragColor quattro 1`] = `
.c0 {
  display: block;
  width: 100%;
  min-height: 18px;
  color: rgb(96,96,96);
  padding-top: 3px;
  background-color: rgb(240,240,240);
  font-family: sans-serif;
  font-size: 11px;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #ffd8bf;
}

<Div
  className="c0"
  dragColor="quattro"
/>
`;

exports[`<SubLabel /> should render its css w/ props dragColor tre 1`] = `
.c0 {
  display: block;
  width: 100%;
  min-height: 18px;
  color: rgb(96,96,96);
  padding-top: 3px;
  background-color: rgb(240,240,240);
  font-family: sans-serif;
  font-size: 11px;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #E0D5E3;
}

<Div
  className="c0"
  dragColor="tre"
/>
`;

exports[`<SubLabel /> should render its css w/ props dragColor uno 1`] = `
.c0 {
  display: block;
  width: 100%;
  min-height: 18px;
  color: rgb(96,96,96);
  padding-top: 3px;
  background-color: rgb(240,240,240);
  font-family: sans-serif;
  font-size: 11px;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #f5ee97;
}

<Div
  className="c0"
  dragColor="uno"
/>
`;

exports[`<SubLabel /> should render its css w/ props dragDisabled 1`] = `
.c0 {
  display: block;
  width: 100%;
  min-height: 18px;
  color: rgb(96,96,96);
  padding-top: 3px;
  background-color: rgb(240,240,240);
  font-family: sans-serif;
  font-size: 11px;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 span {
  color: rgb(240,240,240);
}

<Div
  className="c0"
  dragDisabled={true}
/>
`;
